src/probnum/statespace/sde.py

```diff
@@ -199,12 +210,57 @@ def backward_rv(
 
     # Forward and backward implementation(s)
 
-    def _solve_mde_forward(self, rv, t, dt, _diffusion=1.0):
+    def _solve_mde_forward_classic(self, rv, t, dt, _diffusion=1.0):
         """Solve forward moment differential equations (MDEs)."""
-        mde, y0 = self._setup_vectorized_mde_forward(
+        dim = rv.mean.shape[0]
+        mde, y0 = self._setup_vectorized_mde_forward_classic(
             rv,
             _diffusion=_diffusion,
         )
+
+        sol, new_mean, new_cov = self._solve_mde_forward(mde, y0, t, dt, dim)
+
+        # Useful for backward transitions
+        # Aka continuous time smoothing.
+        sol_mean = lambda t: sol.sol(t)[:dim]
+        sol_cov = lambda t: sol.sol(t)[dim:].reshape((dim, dim))
+
+        return randvars.Normal(mean=new_mean, cov=new_cov), {
+            "sol": sol,
+            "sol_mean": sol_mean,
+            "sol_cov": sol_cov,
+        }
+
+    def _solve_mde_forward_sqrt(self, rv, t, dt, _diffusion=1.0):
+        """Solve forward moment differential equations (MDEs) using a square-root
+        implementation."""
+        dim = rv.mean.shape[0]
+        mde, y0 = self._setup_vectorized_mde_forward_sqrt(
+            rv,
+            _diffusion=_diffusion,
+        )
+
+        sol, new_mean, new_cov_sqrt = self._solve_mde_forward(mde, y0, t, dt, dim)
+        new_cov = new_cov_sqrt @ new_cov_sqrt.T
+
+        # Useful for backward transitions
+        # Aka continuous time smoothing.
+        sol_mean = lambda t: sol.sol(t)[:dim]
+        sol_cov_sqrt = lambda t: sol.sol(t)[dim:].reshape((dim, dim))
```

Comment:
would be great (if this is a cholesky factor) to call this `sol_cov_cholesky` (in line with how we refer to this variable in e.g. `Normal`).

Reply:
this applies to all the code in this module :)

Reply:
>if this is a cholesky factor

The paper does not explicitly state anything regarding `cov_sqrt` being a Cholesky factor. I think `cov_sqrt` should be a Cholesky factor but I am not certain.

Reply:
I think it should (because of the way it is truncated after each step), but you know more about this than I do :) One simple way of finding out would be to run the tests with a simple `assert np.testing.assert_allclose(cov_sqrt, np.tril(cov_sqrt))` and if it passes, it should be a cholesky factor. If so, then renaming these variables would be wonderful, and if not, leave it as is (I am truly sorry to be so petty here with respect to the naming, but this information whether it is lower triangular or not is fairly crucial IMO -- after this we can merge, I promise!! :) )

Reply:
>One simple way of finding out would be to run the tests with a simple assert np.testing.assert_allclose(cov_sqrt, np.tril(cov_sqrt)) and if it passes, it should be a cholesky factor

Yes, this is true by definition. However, I don't see that `cov_sqrt_{ii} > 0` which does need to be true for a Cholesky factor, right?

Reply:
Ah I see. You can transform the factor into a valid cholesky factor though: `S = L1 L1t = L2 Q Qt L2t` and if `Qt` is the (orthogonal) matrix consisting of 1 and -1 on the diagonals, then (in this context here) we can rather loosely switch back between L1 and L2 in order to transform the lower-triangular matrix-sqrt into a Cholesky factor (its diagonals need to be != 0, but that should be no problem here). This is what is done in `probnum.utils.linalg.cholesky_update`, and might be applicable here. What do you think?

Reply:
Oh I see. I haven't thought on that end. My tests suggested that the diagonal entries are indeed positive but I did not see a way to proof this following the algorithm of the paper. I will look into `probnum.utils.linalg.cholesky_update` and report back if that is ok.

Reply:
I see that you re-requested my review -- may I ask what the result here was? :)

Reply:
I am using `probnum.utils.linalg.cholesky_update` to ensure that `cov_cholesky` indeed is a Cholesky factor as you suggested. However, I did not come up with an example where the algorithm produces a triangular matrix with non-positive diagonal entries.